plugins/status.js

```diff
@@ -0,0 +1,244 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const async = require('async');
+
+exports.register = function () {
+    this.outbound = require('../outbound');
+    this.queue_dir = require('../outbound/queue').queue_dir;
+}
+
+exports.hook_capabilities = function (next, connection) {
+
+    if (connection.remote.is_local) {
+        connection.capabilities.push('STATUS');
+    }
+
+    next();
+}
+
+exports.hook_unrecognized_command = function (next, connection, params) {
+    if (params[0] !== 'STATUS') return next();
+    if (!connection.remote.is_local) return next(DENY, "STATUS not allowed remotely");
+
+    this.run(params[1], function (err, result) {
+        if (err) return next(DENY, err.message);
+
+        connection.respond(211, result ? JSON.stringify(result) : "null", () => next(OK));
```

Comment:
You're returning the string "null" here. Typically one would pass a js `null` (no quotes), or an empty string `''` (empty quotes). This string won't evaluate to false like an actual `null` would.

Reply:
It will come out as "211 null" message and thus json parser on other end will convert to `null` also. Due some glitch sending empty string response with `connection.respond(211, '')` is not working and nothing happens.